Library/Homebrew/download_strategy.rb

```diff
@@ -222,27 +222,30 @@ def extract_ref(specs)
 
 class AbstractFileDownloadStrategy < AbstractDownloadStrategy
   def stage
-    case cached_location.compression_type
+    case type = cached_location.compression_type
     when :zip
       with_system_path { quiet_safe_system "unzip", "-qq", cached_location }
       chdir
     when :gzip_only
       with_system_path { buffered_write("gunzip") }
     when :bzip2_only
       with_system_path { buffered_write("bunzip2") }
-    when :gzip, :bzip2, :compress, :tar
+    when :gzip, :bzip2, :xz, :compress, :tar
       tar_flags = "x"
       # Older versions of tar require an explicit format flag
-      if cached_location.compression_type == :gzip
+      if type == :gzip
         tar_flags << "z"
-      elsif cached_location.compression_type == :bzip2
+      elsif type == :bzip2
         tar_flags << "j"
       end
```

Comment:
Does the "older versions of tar" that "require an explicit format flag" happen to be tar on 10.6 or lower? In any case, even if it's unnecessary, IMO it would be better to add a clause for `:xz`:

```ruby
elsif type == :xz
  tar_flags << "J"
end
```

which is arguably safer (no one needs to ask the older versions question I raised above), and at least feels more complete.